File: LEETCODE/Array/26.RemoveDuplicatesFromSortedArray.js
var removeDuplicates = function(nums) {
    const hashMap = new Map();

    for(const n of nums){
        hashMap.set(n, (hashMap.get(n) || 0 )+1) 
    }

    return hashMap.size;
};

const input = [1,1,2];
console.log(removeDuplicates([1,1,2]));

File: LEETCODE/Easy/Array/26_remove_duplicates_from_sorted_array.cpp
#include <iostream>
#include <vector>
using namespace std;

class Solution {
    public:
        int removeDuplicates(vector<int>& nums) {
            int count = 1;
            for(int i =1;i < nums.size();i++){
               if (nums[i] != nums[i-1]){
                    nums[count] = nums[i];
                    count++; 
               } 
            }
            return count;
        }
    };

int main() {
    Solution solution;
    vector<int> nums = {1, 1, 1,1,1,2};
    int newLength = solution.removeDuplicates(nums);
    cout << "New length: " << newLength << endl;
    cout << "Modified array: ";
    for (int i = 0; i < newLength; i++) {
        cout << nums[i] << " ";
    }
    cout << endl;
    return 0;
}